.github/workflows/tests.yaml

```diff
@@ -112,9 +112,20 @@ jobs:
 
     - name: run tests
       run: |
-        cargo nextest run --workspace ${{ env.FEATURES }} --lib --bins --tests --run-ignored ${{ inputs.flaky && 'all' || 'default' }} --no-fail-fast
+        mkdir -p output
+        cargo nextest run --workspace ${{ env.FEATURES }} --lib --bins --tests --run-ignored ${{ inputs.flaky && 'all' || 'default' }} --no-fail-fast --message-format ${{ inputs.flaky && 'libtest-json' || 'human' }} > output/${{ matrix.name }}_${{ matrix.features }}_${{ matrix.rust }}.json
       env:
         RUST_LOG: ${{ runner.debug && 'TRACE' || 'DEBUG'}}
+        NEXTEST_EXPERIMENTAL_LIBTEST_JSON: 1
+
+    - name: upload results
+      if: ${{ failure() && inputs.flaky }}
```

Comment:
why only upload on failure and not always upload?  If it way always uploaded we could also use this info to show statistics of how flaky a test is if we ever felt like doing so.  I think there's very little harm to always upload.  AFAIK github expires artifacts after some time anyway, and I doubt our storage use is a problem right now.

Reply:
since we only use them for failures.. it makes sense to just upload on failure

Reply:
closing to continue in main thread :)